MantidPlot/src/Mantid/InstrumentWidget/InstrumentWindow.cpp

```diff
@@ -1,837 +1,58 @@
-  m_renderTab->m_cylindricalY->setChecked(
-      qs.value("user_params_m_cylindricalY", true).toBool());
-  m_renderTab->m_cylindricalZ->setChecked(
-      qs.value("user_params_m_cylindricalZ", false).toBool());
-  m_renderTab->m_sphericalX->setChecked(
-      qs.value("user_params_m_sphericalX", false).toBool());
-  m_renderTab->m_sphericalY->setChecked(
-      qs.value("user_params_m_sphericalY", false).toBool());
-  m_renderTab->m_sphericalZ->setChecked(
-      qs.value("user_params_m_sphericalZ", false).toBool());
-  m_renderTab->m_sideBySide->setChecked(
-      qs.value("user_params_m_sideBySide", false).toBool());
-
-  // if true assign surfaceType int
-  int surfaceType = 2;
-  if ((qs.value("user_params_Full3d").toBool())) {
-    surfaceType = 0;
-  } else if ((qs.value("user_params_m_cylindricalX").toBool())) {
-    surfaceType = 1;
-  } else if ((qs.value("user_params_m_cylindricalY").toBool())) {
-    surfaceType = 2;
-  } else if ((qs.value("user_params_m_cylindricalZ").toBool())) {
-    surfaceType = 3;
-  } else if ((qs.value("user_params_m_sphericalX").toBool())) {
-    surfaceType = 4;
-  } else if ((qs.value("user_params_m_sphericalY").toBool())) {
-    surfaceType = 5;
-  } else if ((qs.value("user_params_m_sphericalZ").toBool())) {
-    surfaceType = 6;
-  } else if ((qs.value("user_params_m_sideBySide").toBool())) {
-    surfaceType = 7;
-  }
-  setSurfaceType(surfaceType);
-
-  // setAxis can only be initailised after setSurfaceType has been called
-  m_renderTab->setAxis(qs.value("user_param_mAxisCombo").toString());
-  m_renderTab->m_flipCheckBox->setChecked(
-      qs.value("user_params_flipCheckBox", false).toBool());
-
-  colorMapRangeChanged(qs.value("user_param_minVal").toDouble(),
-                       qs.value("user_param_maxVal").toDouble());
-  m_renderTab->nthPowerChanged(qs.value("user_param_nthTerm").toDouble());
-  changeNthPower(qs.value("user_param_nthTerm").toDouble());
-
-  // auto scale after max and min value required as it will
-  // read the above above and uncheck the box
-  m_renderTab->m_autoscaling->setChecked(
-      qs.value("user_params_autoscaling", true).toBool());
-
-  // setting the bin range which is not accessed from InstrumentWindowRenderTab
-  // trouble setting the slider according to bin range?
-  setBinRange(qs.value("user_params_minBinRange").toDouble(),
-              qs.value("user_params_maxBinRange").toDouble());
-
-  qs.endGroup();
-
-  // update according to settings
-  updateInfoText();
-  update();
-}
-
-/**
- * Closes the window if the associated workspace is deleted.
- * @param ws_name :: Name of the deleted workspace.
- * @param workspace_ptr :: Pointer to the workspace to be deleted
- */
 void InstrumentWindow::preDeleteHandle(
     const std::string &ws_name,
     const boost::shared_ptr<Workspace> workspace_ptr) {
-  if (ws_name == m_workspaceName.toStdString()) {
+  if (ws_name == m_instrumentWidget->getWorkspaceName().toStdString()) {
```

Comment:
Would something like `InstrumentWindow::hasWorkspace(const QString& name)` be better?, or better yet, overloading `operator==()` for the comparison.